src/index.ts

```diff
@@ -340,7 +340,18 @@ export function HTPipePreAuthorize<
           ...leftContextOut,
         };
         const rightOut = right.preAuthorize(rightIn);
-        return rightOut;
+        const rightPassed = authorizationPassed(rightOut);
+        if (!rightPassed) {
+          return false;
+        }
+        if (leftOut === true && rightOut === true) {
+          return true;
+        }
```

Comment:
1. misspelled
2. isn't there a typo? it says leftOut.
- [ ] misspell and typo in both